src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/Namespace/SetAzureEventHubNamespace.cs

```diff
@@ -23,126 +23,57 @@ namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
-            Position = 4,
-          ValueFromPipelineByPropertyName = true,
-          HelpMessage = "The eventhub throughput units.",
-          ParameterSetName = AutoInflateParameterSet)]
+        [Parameter(Mandatory = false, ParameterSetName = NamespaceParameterSet, ValueFromPipelineByPropertyName = true, Position = 4,HelpMessage = "The eventhub throughput units.")]
+        [Parameter(Mandatory = false, ParameterSetName = AutoInflateParameterSet, ValueFromPipelineByPropertyName = true, Position = 4, HelpMessage = "The eventhub throughput units.")]
         public int? SkuCapacity { get; set; }
 
-        [Parameter(Mandatory = false,
-            Position = 5,
-            ValueFromPipelineByPropertyName = true,            
-            HelpMessage = "Disable/Enable Namespace.",
-          ParameterSetName = NamespaceParameterSet)]
-        [Parameter(Mandatory = false,
-            Position = 5,
-            ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Disable/Enable Namespace.",
-          ParameterSetName = AutoInflateParameterSet)]
-        public Models.NamespaceState? State { get; set; }
+        [Parameter(Mandatory = false, ParameterSetName = NamespaceParameterSet, ValueFromPipelineByPropertyName = true, Position = 5, HelpMessage = "Disable/Enable Namespace.")]
+        [Parameter(Mandatory = false, ParameterSetName = AutoInflateParameterSet, ValueFromPipelineByPropertyName = true, Position = 5, HelpMessage = "Disable/Enable Namespace.")]
+        public Models.PSNamespaceState? State { get; set; }
 
-        [Parameter(Mandatory = false,
-            Position = 6,
-            ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Hashtables which represents resource Tag.",
-          ParameterSetName = NamespaceParameterSet)]
-        [Parameter(Mandatory = false,
-            Position = 6,
-            ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Hashtables which represents resource Tag.",
-          ParameterSetName = AutoInflateParameterSet)]
+        [Parameter(Mandatory = false, ParameterSetName = NamespaceParameterSet, ValueFromPipelineByPropertyName = true, Position = 6, HelpMessage = "Hashtables which represents resource Tag.")]
+        [Parameter(Mandatory = false, ParameterSetName = AutoInflateParameterSet, ValueFromPipelineByPropertyName = true, Position = 6, HelpMessage = "Hashtables which represents resource Tag.")]
         public Hashtable Tag { get; set; }
 
         /// <summary>
         /// Indicates whether AutoInflate is enabled.
         /// </summary>
-        [Parameter(
-            Mandatory = true,
-            HelpMessage = "Indicates whether AutoInflate is enabled",
-            ParameterSetName = AutoInflateParameterSet
-            )]
+        [Parameter(Mandatory = true, ParameterSetName = AutoInflateParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Indicates whether AutoInflate is enabled")]
```

Comment:
Take out ValueFromPipelineByPropertyName, does not make sense in the case of switch parameters.

Reply:
removed